source/extensions/filters/common/rbac/matchers.cc

```diff
@@ -123,10 +128,21 @@ bool HeaderMatcher::matches(const Network::Connection&,
 }
 
 bool IPMatcher::matches(const Network::Connection& connection, const Envoy::Http::RequestHeaderMap&,
-                        const StreamInfo::StreamInfo&) const {
-  const Envoy::Network::Address::InstanceConstSharedPtr& ip =
-      destination_ ? connection.localAddress() : connection.remoteAddress();
-
+                        const StreamInfo::StreamInfo& info) const {
+  Envoy::Network::Address::InstanceConstSharedPtr ip;
+  switch (type_) {
+  case ConnectionLocal:
+    ip = connection.localAddress();
```

Comment:
change to `info.downstreamLocalAddress()`

If all the addresses are retrieved from the StreamInfo in this function, it's easier to follow exactly which address each of these cases uses.

Reply:
Also updated the PortMatcher to use `info.downstreamLocalAddress()` to be consistent.